adapters/repos/db/shard.go

```diff
@@ -985,23 +1033,72 @@ func (s *Shard) Shutdown(ctx context.Context) error {
 		}
 	}
 
-	// unregister all callbacks at once, in parallel
-	if err = cyclemanager.NewCombinedCallbackCtrl(0, s.index.logger,
-		s.cycleCallbacks.compactionCallbacksCtrl,
-		s.cycleCallbacks.flushCallbacksCtrl,
-		s.cycleCallbacks.vectorCombinedCallbacksCtrl,
-		s.cycleCallbacks.geoPropsCombinedCallbacksCtrl,
-	).Unregister(ctx); err != nil {
-		return err
-	}
-
 	if err = s.store.Shutdown(ctx); err != nil {
 		return errors.Wrap(err, "stop lsmkv store")
 	}
 
 	return nil
 }
 
+func (s *Shard) preventShutdown() (release func(), err error) {
+	s.shutdownLock.RLock()
+	defer s.shutdownLock.RUnlock()
+
+	if s.shut {
```

Comment:
haven't found where s.shut is set to true...

Reply:
inside `Shard::checkEligibleForShutdown`